connect/tls.go

```diff
@@ -111,6 +112,17 @@ func devTLSConfigFromFiles(caFile, certFile,
 	return cfg, nil
 }
 
+// rawSubjectsFromCertPool uses the reflect package to get rawSubject fields.
+func rawSubjectsFromCertPool(cp *x509.CertPool) []string {
+	subjectsIter := reflect.ValueOf(cp).Elem().FieldByName("byName").MapRange()
+	subjects := []string{}
+	for subjectsIter.Next() {
+		k := subjectsIter.Key()
+		subjects = append(subjects, k.String())
+	}
+	return subjects
+}
+
```

Comment:
Sadly, this is the first time I've used `reflect`. Something about this seems like it needs a safety check but I didn't dig too deep to figure out where.